falcon_multiqc/commands/save.py

```diff
@@ -116,21 +119,57 @@ def save_sample(directory, sample_metadata, session, cohort_description, batch_d
                 session.add(sample_row)
                 session.flush()
                 samples[sample_name] = sample_row.id
+                if type not in types[cohort_id]:
+                    types[cohort_id].append(type)
+
+            # Enter sample counts for batch/cohort.
+            for cohort_id in batches:
+                cohort_count = session.query(Cohort).filter(Cohort.id == cohort_id).one().sample_count
+                if cohort_count == None:
+                    cohort_count = 0
+                for batch_name in batches[cohort_id]:
+                    # Count number of samples in given batch.
+                    batch_count = session.query(Sample).join(Batch, Batch.id == Sample.batch_id).\
+                    filter(Batch.batch_name == batch_name, Sample.cohort_id == cohort_id).count()
+                    # Update Batch count column.
+                    session.query(Batch).filter(Batch.batch_name == batch_name, Batch.cohort_id == cohort_id).one().sample_count = batch_count
+                    cohort_count += batch_count
+                session.query(Cohort).filter(Cohort.id == cohort_id).one().sample_count = cohort_count
+                
+            # Update cohort tables with types if needed
+            for cohort_id in types:
+                old_type_list = session.query(Cohort).filter(Cohort.id == cohort_id).one().type
+                if old_type_list == None:
+                    session.query(Cohort).filter(Cohort.id == cohort_id).one().type = ','.join(types[cohort_id])
+                    continue
+                # Update cohort with new type info.
+                new_type_list = []
+                for type in types[cohort_id]:
+                    if type not in old_type_list:
+                        new_type_list.append(type)
+                new_type_list = ','.join(new_type_list)
+                new_type_list = new_type_list + old_type_list
+                session.query(Cohort).filter(Cohort.id == cohort_id).one().type = new_type_list
 
             multiqc_data_json = json.load(multiqc_data)
 
             for tool in multiqc_data_json["report_saved_raw_data"]:
```

Comment:
Do we want to skip "multiqc_general_stats" here? I'm actually not sure, what do you think? Currently we save it.

Reply:
what does it actually do?